collector/mysql_user.go

```diff
@@ -0,0 +1,362 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Scrape `mysql.user`.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const mysqlUserQuery = `
+		  SELECT
+		    user,
+		    host,
+		    IF( Select_priv = 'N', 0, 1 ) AS select_priv,
```

Comment:
It's better to handle this in the code, rather than create a larger messy SELECT statement. There's a `parseStatus()` helper function in collector.go. You could create a `parsePrivilege()` function to deal with the Y/N here.